### PR TITLE
Adopt ElasticsearchConnector from `search-client-connect` to create and configure ES clients

### LINKED ISSUE #22745: Adopt `search-client-connect` to create and configure ES/OS clients

### PR DESCRIPTION
**Pull request by mustafadagher**
## Description

This creates less duplicate code and adds SSL configuration out of the box.

## Related issues

closes #22745

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/config/ElasticsearchProperties.java
Comment (mustafadagher) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/config/ElasticsearchProperties.java:14]: I thought about the same. No good reason other than to keep the config hierarchy as is of it right now. It could be very much also a composed field.

Comment (EuroLew) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/config/ElasticsearchProperties.java:14]: ok I'll approve as long as this becomes a class field before you merge.

**Review by EuroLew — APPROVED** [review]

**Pull request merged**
